packages/gatsby/src/schema/__tests__/infer-graphql-input-type-test.js

```diff
@@ -483,3 +483,63 @@ describe(`GraphQL Input args`, () => {
     expect(result).toMatchSnapshot()
   })
 })
+
+describe(`filtering on linked nodes`, () => {
+  let types
+  beforeEach(() => {
+    const { store } = require(`../../redux`)
+    types = [
+      {
+        name: `Child`,
+        nodeObjectType: new GraphQLObjectType({
+          name: `Child`,
+          fields: inferObjectStructureFromNodes({
+            nodes: [{ id: `child_1`, hair: `brown`, height: 101 }],
+            types: [{ name: `Child` }],
+          }),
+        }),
+      },
+      {
+        name: `Pet`,
+        nodeObjectType: new GraphQLObjectType({
+          name: `Pet`,
+          fields: inferObjectStructureFromNodes({
+            nodes: [{ id: `pet_1`, species: `dog` }],
+            types: [{ name: `Pet` }],
+          }),
+        }),
+      },
+    ]
+
+    store.dispatch({
+      type: `CREATE_NODE`,
+      payload: { id: `child_1`, internal: { type: `Child` }, hair: `brown` },
+    })
+    store.dispatch({
+      type: `CREATE_NODE`,
+      payload: { id: `child_2`, internal: { type: `Child` }, hair: `blonde`, height: 101 },
+    })
+    store.dispatch({
+      type: `CREATE_NODE`,
+      payload: { id: `pet_1`, internal: { type: `Pet` }, species: `dog` },
+    })
+  })
+
+  it(`filters on linked nodes via id`, async () => {
+    let result = await queryResult(
+      [{ linked___NODE: `child_2`, foo: `bar` }, { linked___NODE: `child_2`, foo: `baz` }],
```

Comment:
Does filtering actually work with this? Or this only adds linked node fields to input schema?

It doesn't seem you test if filtering on linked nodes works here (as `linked.hair` in all supplied nodes is actually `blonde` so nothing will be filtered out).

Reply:
That's actually a test I used to verify that the query will actually return an array of results. I could send in another test in a short while.